Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionActionCocoa.mm

```diff
@@ -209,6 +213,9 @@ - (void)invalidateIntrinsicContentSize
         return m_customPopupPath;
 
     if (m_tab)
+        return extensionContext()->getAction(m_tab->window().get())->popupPath();
+
+    if (m_window)
         return extensionContext()->defaultAction().popupPath();
```

Comment:
Should this be consulting window in any way?

Reply:
It does, the fallback is custom value -> window -> default.

The if checks for tab and window happen after the custom value, so they are the fallback cases to the next level. In a tab action (if m_tab) and the tab didn't have a custom value, it checks the window. Then in the window action (if m_window) since it didn't have a custom value, it falls back to the default. And default is !m_tab and !m_window, so it falls back to the default from the manifest if it didn't have a custom value.